rewrite/rewrite/markers.py

```diff
@@ -59,6 +61,31 @@ def find_first(self, type: type):
     def find_all(self, type: type):
         return [m for m in self.markers if isinstance(m, type)]
 
+    def compute_if(self, condition: Callable[[Marker], bool], remap_fn: Callable[[Marker], Marker]) -> Markers:
```

Comment:
This function seems a bit dangerous in its current form, as it will effectively remove any other markers, if at least one marker was updated. As the function isn't used outside of `compute_by_type` I think we should completely remove it to not increase the API surface area before we have an actual use case. Instead, I think `compute_by_type` should use `list_map`:

```py
        return self.with_markers(list_map(lambda m: remap_fn(m) if isinstance(m, cls) else m, self.markers))
```